internal/aws/k8s/k8sclient/node.go

```diff
@@ -222,17 +274,33 @@ func transformFuncNode(obj any) (any, error) {
 	if !ok {
 		return nil, fmt.Errorf("input obj %v is not Node type", obj)
 	}
-	info := new(nodeInfo)
-	info.name = node.Name
-	info.capacity = node.Status.Capacity
-	info.allocatable = node.Status.Allocatable
-	info.conditions = []*NodeCondition{}
+	info := new(NodeInfo)
+	info.Name = node.Name
+	info.Capacity = node.Status.Capacity
+	info.Allocatable = node.Status.Allocatable
+	if instanceType, ok := node.Labels[instanceTypeLabelKey]; ok {
+		info.InstanceType = instanceType
+	} else {
+		// fallback for compatibility with k8s versions older than v1.17
+		// https://kubernetes.io/docs/reference/labels-annotations-taints/#beta-kubernetes-io-instance-type-deprecated
+		if instanceType, ok := node.Labels[instanceTypeLabelKeyBeta]; ok {
+			info.InstanceType = instanceType
+		}
+	}
+	info.Conditions = []*NodeCondition{}
 	for _, condition := range node.Status.Conditions {
-		info.conditions = append(info.conditions, &NodeCondition{
+		info.Conditions = append(info.Conditions, &NodeCondition{
 			Type:   condition.Type,
 			Status: condition.Status,
 		})
 	}
+
+	if sageMakerHealthStatus, ok := node.Labels[SageMakerNodeHealthStatus.String()]; ok {
+		info.Labels = make(map[Label]int8)
+		if condition, ok := k8sutil.ParseString(sageMakerHealthStatus); ok {
+			info.Labels[SageMakerNodeHealthStatus] = condition
```

Comment:
It looks like this map will have a max of one label for the health status and not have any other labels.
Does this even have to be a map in that case? Cant this just be a string called `sageMakerNodeHealthStatus` as part of `NodeInfo` and itll be empty for non hyperpod nodes. Callers can check for empty string or ret pointer and check for nil.

You can probably remove the `captureOnlyNodeLabelInfo` as well at that point?

Reply:
Calling it something generic like `NodeToLabelsMap` gives readers an expectation that all the labels set on the node are accessible via this map.

Reply:
The map are part of an allowlist that uses an ENUM instead of strings to store the label keys/values, any additional labels to be stored will have to be made into an ENUM, thus we will continue to use this pattern for now.